src/test/fuzz/transaction.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <coins.h>
+#include <consensus/tx_check.h>
+#include <consensus/tx_verify.h>
+#include <consensus/validation.h>
+#include <core_io.h>
+#include <core_memusage.h>
+#include <policy/policy.h>
+#include <policy/settings.h>
+#include <primitives/transaction.h>
+#include <streams.h>
+#include <test/fuzz/fuzz.h>
+#include <util/rbf.h>
+#include <validation.h>
+#include <version.h>
+
+#include <cassert>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    CDataStream ds(buffer, SER_NETWORK, INIT_PROTO_VERSION);
+    try {
+        int nVersion;
+        ds >> nVersion;
+        ds.SetVersion(nVersion);
+    } catch (const std::ios_base::failure& e) {
+        return;
+    }
+    const CTransaction tx = [&] {
+        try {
+            return CTransaction(deserialize, ds);
+        } catch (const std::ios_base::failure& e) {
+            return CTransaction();
```

Comment:
why not return early here. Seems wasteful to map all deserialization failures to an default constructed tx and then run all the checks with that.

Also, `TRANSACTION_DESERIALIZE` can be removed from the existing fuzzer now, as the new fuzz test includes this dumb code path now?

Reply:
Good points. Feedback addressed. Please re-review :)